proto/kuksa/val/v2/types.proto

```diff
@@ -147,18 +159,20 @@ message ValueRestrictionString {
   repeated string allowed_values = 1;
 }
 
+// Used to indicate status of a signal in Databroker
+// This is given as an alternative to Value, so if there is a valid value
+// ValueFailure shall not be specified.
+//
+// Scenarios where a signal does not exist or a user does not have access to a signal
+// shall typically not be represented as a ValueFailure, as those secnarios
+// should result in a GRPC error code being returned to the caller.
 enum ValueFailure {
   // Unspecified value failure, reserved for gRPC backwards compatibility
   // (see https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum)
   UNSPECIFIED    = 0;
-  // The signal is known and provided, but doesn't have a valid value
-  INVALID_VALUE  = 1;
   // The signal is known, but no value is provided currently
+  // Can theoretically also be used in write request to "delete" a value
```

Comment:
Should we possible for each one indicate in which direction they can/shall be used? I can see a use-case for sending `NOT_PROVIDED` towards databroker, but I see no reason to send `INTERNAL_ERROR` to Databroker.

We can discuss if we want to keep  `INTERNAL_ERROR`. For single calls I see no need as we can return a GRPC error code. For calls that shall return multiple signals (like `GetValues`) it could make sense to be able to return other signals. For streaming it makes sense if we want to continue to keep the stream open upon an internal error to be able to continue subscription for other signals. But internal error should never happen right?

Reply:
You are right, this still needs to be discussed. We still wanted to clean up this enum.